ui/components/multichain/create-named-snap-account/create-named-snap-account.tsx

```diff
@@ -0,0 +1,83 @@
+import React from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory } from 'react-router-dom';
+import {
+  getNextAvailableAccountName as getNextAvailableAccountNameFromController,
+  setAccountLabel,
+} from '../../../store/actions';
+import { CreateAccount } from '..';
+import { Box, ModalHeader } from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { getMostRecentOverviewPage } from '../../../ducks/history/history';
+import { getSnapRoute } from '../../../helpers/utils/util';
+
+export type CreateNamedSnapAccountProps = {
+  /**
+   * Callback called once the account has been created
+   */
+  onActionComplete: (completed: boolean, reject?: boolean) => Promise<void>;
+
+  /**
+   * Address of the account to create
+   */
+  address: string;
+
+  /**
+   * Snap Id
+   */
+  snapId: string;
+
+  /**
+   * Suggested account name from the snap
+   */
+  snapSuggestedAccountName?: string;
+};
+
+export const CreateNamedSnapAccount: React.FC<CreateNamedSnapAccountProps> = ({
+  onActionComplete,
+  address,
+  snapId,
+  snapSuggestedAccountName,
+}) => {
+  const t = useI18nContext();
+  const history = useHistory();
+  const dispatch = useDispatch();
+
+  const mostRecentOverviewPage = useSelector(getMostRecentOverviewPage);
+
+  const onCreateAccount = async (name: string) => {
+    dispatch(setAccountLabel(address, name));
+    await onActionComplete(true);
+  };
+
+  const getNextAccountName = async (): Promise<string> => {
+    const defaultAccountName =
+      await getNextAvailableAccountNameFromController();
```

Comment:
This function (on the controller side at least), can take a keyring type as argument, which allows to generate the next account name for a given keyring. So I'd suggest you to try to use `KeyringTypes.snap` and see if that works.

This way we would get back to the initial naming schema "Snap Account N". WDYT?

Reply:
Nice one! I didn't realize we could do that :) I updated actions.ts and passing the snap keyring into the function in this commit [352d9ae](https://github.com/MetaMask/metamask-extension/pull/25191/commits/352d9ae33c55692a2305c10da3a0fcb3757f911a)